core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/QuarkusControllerConfigurationBuildStep.java

```diff
@@ -520,6 +527,31 @@ private static DependentResourceSpecMetadata createDependentResourceSpec(
         return spec;
     }
 
+    private static void failOnMissingResourceIDProvider(String dependentTypeName, String resourceTypeName, IndexView index) {
+        if (!index.getAllKnownImplementations(DotName.createSimple(ResourceIDMapper.class)).isEmpty()) {
```

Comment:
@metacosm do we know if the `ResourceIDMapper` is in use in some of the event sources?